src/protocol/ldap/org/apache/jmeter/protocol/ldap/sampler/TrustAllSocket.java

```
@@ -0,0 +1,121 @@
+import javax.net.ssl.X509ExtendedTrustManager;
+
+public class TrustAllSocket extends SocketFactory {
+    private static final AtomicReference<TrustAllSocket> defaultFactory = new AtomicReference<>();
+
+    private static SSLSocketFactory sf;
+
+    public TrustAllSocket() {
+        //KeyStore keyStore = ... /* Get a keystore containing the self-signed certificate) */
+        TrustManager[] trustAllCerts = new TrustManager[] {new X509ExtendedTrustManager() {
+            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+            public void checkClientTrusted(X509Certificate[] certs, String authType) {
+            }
+            public void checkServerTrusted(X509Certificate[] certs, String authType) {
+            }
+            @Override
+            public void checkClientTrusted(X509Certificate[] arg0, String arg1, Socket arg2)
+                throws CertificateException {
+                // TODO Auto-generated method stub
+            }
+            @Override
+            public void checkClientTrusted(X509Certificate[] arg0, String arg1, SSLEngine arg2)
+                throws CertificateException {
+                // TODO Auto-generated method stub
+            }
+            @Override
+            public void checkServerTrusted(X509Certificate[] arg0, String arg1, Socket arg2)
+                throws CertificateException {
+                // TODO Auto-generated method stub
+            }
+            @Override
+            public void checkServerTrusted(X509Certificate[] arg0, String arg1, SSLEngine arg2)
+                throws CertificateException {
+                // TODO Auto-generated method stub
+            }
+        }};
+
+        SSLContext ctx = null;
+        try {
+            ctx = SSLContext.getInstance("TLS");
+        } catch (NoSuchAlgorithmException e1) {
+            // TODO Auto-generated catch block
+            e1.printStackTrace();
```

Logging is better than printStackTrace(). IMO let it throw.